src/main/kotlin/team/jeonghokim/zakbu/global/security/jwt/JwtTokenProvider.kt

```diff
@@ -0,0 +1,86 @@
+package team.jeonghokim.zakbu.global.security.jwt
+
+import io.jsonwebtoken.Claims
+import io.jsonwebtoken.Jwts
+import io.jsonwebtoken.SignatureAlgorithm
+import jakarta.servlet.http.HttpServletRequest
+import org.springframework.security.authentication.UsernamePasswordAuthenticationToken
+import org.springframework.security.core.Authentication
+import org.springframework.security.core.userdetails.UserDetails
+import org.springframework.stereotype.Component
+import team.jeonghokim.zakbu.domain.auth.domain.RefreshToken
+import team.jeonghokim.zakbu.domain.auth.domain.exception.ExpiredJwtException
+import team.jeonghokim.zakbu.domain.auth.domain.exception.InvalidJwtException
+import team.jeonghokim.zakbu.domain.auth.domain.repository.RefreshTokenRepository
+import team.jeonghokim.zakbu.global.security.auth.AuthDetailsService
+import java.util.*
+import javax.crypto.spec.SecretKeySpec
+
+@Component
+class JwtTokenProvider(
+    private val jwtProperties: JwtProperties,
+    private val authDetailsService: AuthDetailsService,
+    private val refreshTokenRepository: RefreshTokenRepository
+) {
+    private val secretKeySpec = SecretKeySpec(
+        jwtProperties.secretKey.toByteArray(),
+        SignatureAlgorithm.HS256.jcaName)
+
+    private fun generateToken(email: String, type: String, exp: Long): String {
+        return Jwts.builder()
+            .signWith(secretKeySpec)
+            .setSubject(email)
+            .setHeaderParam("type", type)
+            .setIssuedAt(Date())
+            .setExpiration(Date(System.currentTimeMillis() + exp * 1000))
+            .compact()
+    }
+
+    fun generateAccessToken(email: String): String {
+        return generateToken(email, "access", jwtProperties.accessExp)
+    }
+
+    fun generateRefreshToken(email: String): String {
+        val refreshToken = generateToken(email, "refresh", jwtProperties.refreshExp)
```

Comment:
`companion object`를 사용해서 `ACCESS`와 `REFESH`를 만든 후 하드 코딩을 최소화하면 좋을 것 같아요!

```suggestion
        val refreshToken = generateToken(email, REFRESH, jwtProperties.refreshExp)
```